uint256.go

```diff
@@ -198,9 +198,29 @@ func (z *Int) AddMod(x, y, m *Int) *Int {
 	if m.IsZero() {
 		return z.Clear()
 	}
-	if z == m { // z is an alias for m  // TODO: Understand why needed and add tests for all "division" methods.
+	if z == m { // z is an alias for m and will be overwritten by AddOverflow before m is read
 		m = m.Clone()
 	}
+
+	// Fast path for m >= 2^192, with x and y at most slightly bigger than m.
+	// This is always the case when x and y are already reduced modulo such m.
+
+	if (m[3] != 0) && (m[3] >= x[3]) && (m[3] >= y[3]) {
```

Comment:
```suggestion
	if (m[3] != 0) && (x[3] <= m[3]) && (y[3] <= m[3]) {
```

Reply:
If you and others feel that this is more readable, then it's ok with me.

Reply:
For me it is because I think of `x < m` and `y < m` as preconditions to efficient ADDMOD implementation.